src/main/java/org/embulk/output/sftp/SftpFileOutput.java

```diff
@@ -250,84 +282,54 @@ private String getOutputFilePath()
-            }
-        };
-        try {
-            return withConnectionRetry(retriable);
-        }
-        catch (Exception e) {
-            throw (FileSystemException) e;
-        }
-    }
-
-    private OutputStream newSftpOutputStream(final FileObject file)
-            throws FileSystemException
-    {
-        Retriable<OutputStream> retriable = new Retriable<OutputStream>() {
-            public OutputStream execute() throws FileSystemException
-            {
-                return file.getContent().getOutputStream();
-            }
-        };
-        try {
-            return withConnectionRetry(retriable);
-        }
-        catch (Exception e) {
-            throw (FileSystemException) e;
-        }
+        return retryExecutor()
+                .withRetryLimit(maxConnectionRetry)
+                .withInitialRetryWait(500)
+                .withMaxRetryWait(30 * 1000)
+                .runInterruptible(new Retryable<FileObject>() {
+                    @Override
+                    public FileObject call() throws FileSystemException, RetryGiveupException
+                    {
+                        FileObject file = manager.resolveFile(sftpUri.toString(), fsOptions);
+                        if (file.getParent().exists()) {
+                            logger.info("parent directory {} exists there", file.getParent().getPublicURIString());
+                        }
+                        else {
+                            logger.info("trying to create parent directory {}", file.getParent().getPublicURIString());
+                            file.getParent().createFolder();
+                        }
+                        return file;
+                    }
+
+                    @Override
+                    public boolean isRetryableException(Exception exception)
+                    {
+                        return true;
+                    }
+
+                    @Override
+                    public void onRetry(Exception exception, int retryCount, int retryLimit, int retryWait)
+                            throws RetryGiveupException
+                    {
+                        String message = String.format("SFTP resolve file request failed. Retrying %d/%d after %d seconds. Message: %s",
+                                retryCount, retryLimit, retryWait / 1000, exception.getMessage());
+                        if (retryCount % 3 == 0) {
+                            logger.warn(message, exception);
+                        }
+                        else {
+                            logger.warn(message);
+                        }
+                    }
+
+                    @Override
+                    public void onGiveup(Exception firstException, Exception lastException)
+                            throws RetryGiveupException
+                    {
+                    }
+                });
     }
 
     private Function<LocalFile, String> localFileToPathString()
```

Comment:
If the `localFileToPathString` method is not called any more, please remove the method definition.